expression/builtin_json.go

```diff
@@ -17,13 +17,27 @@ import (
 	"github.com/juju/errors"
 	"github.com/pingcap/tidb/ast"
 	"github.com/pingcap/tidb/context"
-	"github.com/pingcap/tidb/mysql"
 	"github.com/pingcap/tidb/sessionctx/variable"
 	"github.com/pingcap/tidb/util/types"
 	"github.com/pingcap/tidb/util/types/json"
 	"github.com/pingcap/tipb/go-tipb"
 )
 
+const (
+	// decimal4CastJSONDirectly is used for really cast to JSON, which means for
+	// string we should parse it into JSON but not use that as primitive.
+	// For example, CAST('{}' as JSON) will return JSON::Str('{}').
+	decimal4CastJSONDirectly int = 0
+	// decimal4CastJSONPostWrapped is used for post-wrapped cast to JSON, which
+	// means for string we should use that as primitive but not parse it.
+	// For example, CAST('{}' as JSON) will return JSON::Object(empty).
+	decimal4CastJSONPostWrapped int = -1
+
+	// jsonTypeFlen is the Flen of JSON_TYPE builtin function.
+	// It comes from length of UNSIGNED INTEGER.
+	jsonTypeFlen int = 51
```

Comment:
no need to declare it as a global constant, since it only used in `jsonTypeFunctionClass`